index.js

```diff
@@ -35,12 +35,19 @@ module.exports = function(content) {
 
 	var publicPath = "__webpack_public_path__ + " + JSON.stringify(url);
 
-	if (config.publicPath) {
+	if (query.publicName) {
+		var publicUrl = loaderUtils.interpolateName(this, query.publicName, {
+			context: config.context || this.options.context,
+			content: content,
+			regExp: config.regExp
+		});
+		publicPath = JSON.stringify(publicUrl);
+	} else if (config.publicPath) {
 		// support functions as publicPath to generate them dynamically
 		publicPath = JSON.stringify(
-				typeof config.publicPath === "function" 
-				 ? config.publicPath(url) 
-				 : config.publicPath + url
+				typeof config.publicPath === "function"
+				 ? config.publicPath(url)
+				 : publicUrl
```

Comment:
I don't think this is going to behave as intended. `publicUrl` isn't defined here, it was initialized only inside the `if` branch of the  `if...else`. Shouldn't this be the original code (`config.publicPath + url`)?